packages/syft/src/syft/service/dataset/dataset.py

```diff
@@ -597,6 +601,9 @@ def add_asset(
             message=f"Asset '{asset.name}' added to '{self.name}' Dataset."
         )
 
+    def replace_asset(self, asset: CreateAsset):
```

Comment:
I think we wanted it to be at the Dataset class instead, where it calls the dataset update service. 
The use case is to replace an already uploaded asset.